Library/Homebrew/unpack_strategy/dmg.rb

```diff
@@ -87,10 +87,31 @@ def eject(verbose: false)
         return unless path.exist?
 
         if tries > 1
-          system_command! "diskutil",
-                          args:         ["eject", path],
-                          print_stderr: false,
-                          verbose:      verbose
+          disk_info = system_command!(
+            "diskutil",
+            args:         ["info", "-plist", path],
+            print_stderr: false,
+            verbose:      verbose,
+          )
+
+          # For HFS, just use <mount-path>
+          # For APFS, find the <physical-store> corresponding to <mount-path>
+          eject_paths = if disk_info.success?
+            disk_info.plist
+                     .fetch("APFSPhysicalStores", [])
+                     .map { |store| store["APFSPhysicalStore"] }
+                     .compact
+                     .presence || [path]
+          else
+            [path]
+          end
```

Comment:
This should be outside of the `retry` loop.

Reply:
Okay. I wasn't sure about `diskutil unmount` behavior, but it seems documentation does specify device/partition.

The behavior doesn't really seem to change though as it looks like unmount only affects Volumes, which only exist on virtual disk for APFS.

Reply:
From what I tried only `unmountDisk` worked for `diskXsY` IDs.

Reply:
Trying with calibre DMG, it works on Big Sur:
```console
❯ diskutil unmountDisk /Volumes/calibre-5.14.0
Unmount of all volumes on disk5 was successful
❯ diskutil unmountDisk /dev/disk4
Unmount of all volumes on disk4 was successful
❯ diskutil unmountDisk /dev/disk4s1
Unmount of all volumes on disk4 was successful
❯ diskutil unmountDisk disk4
Unmount of all volumes on disk4 was successful
❯ diskutil unmountDisk disk4s1
Unmount of all volumes on disk4 was successful

```

As mentioned in other review though, there is no real difference between:
- current looped logic over `diskutil unmount` and 
- `diskutil unmountDisk`

And, for APFS, there doesn't appear to be a difference between:
- current looped logic over `diskutil unmount` and
- `diskutil unmountDisk <physical-device>`

The only issue for APFS is `diskutil eject` logic, which needs to be run on physical store to be similar to how HFS behaves.

Due to this, I think leaving the current implementation should be fine as it only changes that part of code (`diskutil eject`) that is impacted by new filesystem.